pandas/tools/merge.py

```diff
@@ -199,10 +206,50 @@ def get_result(self):
         typ = self.left._constructor
         result = typ(result_data).__finalize__(self, method='merge')
 
+        if self.indicator:
+            result = self._indicator_post_merge(result)
+
         self._maybe_add_join_keys(result, left_indexer, right_indexer)
 
         return result
 
+    def _indicator_pre_merge(self, left, right):
+
+        if not isinstance(self.indicator, bool) and not isinstance(self.indicator, 
```

Comment:
@jreback in previous comments you suggested I don't need this (since we only get to this if `self.indicator is not False`), but I think I still do in case someone passed an integer or float for some reason. That right?